### ISSUE TITLE
Building 1.3 in a Docker Container failed

### ISSUE THREAD

**Issue by mschae** [LeToteTeam/kiosk_system_rpi3#30]
I am trying to build a project very much like your example on Docker.

This is the Dockerfile:
```Dockerfile
FROM elixir:1.6-otp-21

RUN apt-get update \
 && \
 apt-get -y install \
 build-essential \
 automake \
 autoconf \
 git \
 squashfs-tools \
 ssh-askpass \
 erlang-dev \
 qtwebengine5-dev \
 qtmultimedia5-dev \
 qt5-default

RUN wget https://github.com/fhunleth/fwup/releases/download/v1.2.5/fwup_1.2.5_amd64.deb
RUN dpkg -i fwup_1.2.5_amd64.deb

RUN mix local.hex --force
RUN mix local.rebar --force

RUN mix archive.install hex nerves_bootstrap --force

WORKDIR /app

COPY mix.exs mix.lock* /app/

ENV MIX_TARGET=rpi3

RUN mix deps.get

COPY . /app

CMD mix firmware
```

And this is the error message (stripped to the error):
```
==> webengine_kiosk
cd src && qmake kiosk.pro
/root/.nerves/artifacts/kiosk_system_rpi3-portable-1.3.0/staging/mkspecs/features/toolchain.prf:69: system(execute) requires one or two arguments.
Project ERROR: Cannot run target compiler '/root/.nerves/artifacts/nerves_toolchain_arm_unknown_linux_gnueabihf-linux_x86_64-1.1.0/bin/arm-unknown-linux-gnueabihf-g++'. Output:
===================
===================
Maybe you forgot to setup the environment?
Makefile:7: recipe for target 'src/Makefile' failed
make: *** [src/Makefile] Error 3
could not compile dependency :webengine_kiosk, "mix compile" failed. You can recompile this dependency with "mix deps.compile webengine_kiosk", update it with "mix deps.update webengine_kiosk" or clean it with "mix deps.clean webengine_kiosk"
==> whyit_kiosk
** (Mix) Could not compile with "make" (exit status: 2).
You need to have gcc and make installed. If you are using
Ubuntu or any other Debian-based system, install the packages
"build-essential". Also install "erlang-dev" package if not
included in your Erlang/OTP version. If you're on Fedora, run
"dnf group install 'Development Tools'".
```

Thank you,
Michael

**Comment by mobileoverlord** [comment]
I was able to reproduce the issue, but for what its worth, you don't need to also install qt5-webengine and such on the container. The only host requirement for this to compile is that qmake is installed. It will also compile on Mac.

**Comment by mschae** [comment]
Yeah I figured we only need qt5 et al on the target system but I saw the qmake error and figured to just install everything that `webengine_kiosk` suggested. :)

Any way I can help fixing the issue?

**Comment by mschae** [comment]
Hey,

sorry for nagging but any updates on this? Unfortunately I can't install OTP21 on my Mac right now (asdf-vm/asdf-erlang#82) and generally we try to have everything runnable in Docker for CI reasons.

We'd love to use the kiosk on our IT campus screens here.

Thanks,
Michael

**Comment by mobileoverlord** [comment]
Unfortunately, I do not have any time to debug this issue. We've been pretty busy preparing for ElixirConf. Its a little baffling that this works on Mac and Linux natively, but is having trouble in the container. It appears that container is debian stretch based, I would try to build a fresh image, or better yet, base it off the docker image for `nervesproject/nerves_system_br:1.4.3`

**Comment by fhunleth** [comment]
@mschae Sorry, I'm only just now seeing this now. I think that you have an incompatible `qmake` version. I've learned that the qmake spec files have some information in them that doesn't work with old versions of qmake. For example, this version of qmake works for me:

```sh
$ qmake --version
QMake version 3.1
Using Qt version 5.9.5 in /usr/lib/x86_64-linux-gnu
```
Newer versions work as well. I suspect that you have an older version.

**Comment by mschae** [comment]
@fhunleth Sorry only got around to this now. 

Here is my output:

```
QMake version 3.0
Using Qt version 5.7.1 in /usr/lib/x86_64-linux-gnu
```

It's what is in debian:stretch's apt repository. Do I need to install a more recent version?

**Comment by fhunleth** [comment]
Yes, we use Qt 5.11.1 in Nerves and we found out during training that older versions like the one you have give hard to understand errors. I had thought that the qmake metadata was fairly stable between versions, but apparently not. My current "best idea" for fixing this is to download the opensource Qt distro from http://qt.io/.

@mobileoverlord If we ever get to adding host artifacts beyond the toolchains, adding a `qmake` one would really be helpful. Also, we use the Qt metaobject compiler (moc) and the resource compiler (qrc, I think) on the host, so switching away from `qmake` to `cmake` or something else doesn't fix the underlying issue.

**Comment by OldhamMade** [comment]
Was a workaround ever found for this? I'm developing on a Mac and I'm using docker to target an rpi3, but I can't get past the error.

**Comment by bjufre** [comment]
Trying to build kiosk system (for RPI4) I got this error running `mix compile`.
<img width="1396" alt="Screenshot 2019-09-16 at 14 17 48" src="https://user-images.githubusercontent.com/22913276/64975636-cb1c0b80-d8af-11e9-897c-45a82b2ab2fe.png">

I asked in Slack and @fhunleth pointed out that might be because of running out of memory or something similar when trying to compile Chromium.

So instead of running `mix compile` directly from Mac iTerm, I jumped into the Docker container and run `make clean && make` and the final output (max I could screenshot from the running the compilation) was:
<img width="1920" alt="Screenshot 2019-09-16 at 18 32 17" src="https://user-images.githubusercontent.com/22913276/64975908-61503180-d8b0-11e9-8546-2eccfa0605f6.png">


Any help would be very much welcome as this is for a client's project, thank you very much!

**Comment by zhulinpinyu** [comment]
@bjufre  you need setup following in `nerves_defconfig`
```
BR2_PACKAGE_QT5WEBENGINE_PROPRIETARY_CODECS=y
BR2_PACKAGE_QT5WEBENGINE_ALSA=y
```
